lib/benchee/formatters/html.ex

```diff
@@ -1,170 +1,70 @@
-  end
-
-  defp render_footer do
-    footer(%{
-      benchee: Application.spec(:benchee, :vsn),
-      benchee_html: Application.spec(:benchee_html, :vsn)
-    })
-  end
-
   @moduledoc """
   Functionality for converting Benchee benchmarking results to an HTML page
   with plotly.js generated graphs and friends.
 
   ## Examples
 
-      list = Enum.to_list(1..10_000)
-      map_fun = fn(i) -> [i, i * i] end
+    list = Enum.to_list(1..10_000)
+    map_fun = fn(i) -> [i, i * i] end
 
-      Benchee.run(%{
-        "flat_map"    => fn -> Enum.flat_map(list, map_fun) end,
-        "map.flatten" => fn -> list |> Enum.map(map_fun) |> List.flatten end
+    Benchee.run(
+      %{
+        "flat_map" => fn -> Enum.flat_map(list, map_fun) end,
+        "map.flatten" => fn -> list |> Enum.map(map_fun) |> List.flatten() end
       },
-        formatters: [
-          &Benchee.Formatters.HTML.output/1,
-          &Benchee.Formatters.Console.output/1
-        ],
-        formatter_options: [html: [file: "samples_output/flat_map.html"]],
-      )
-
+      formatters: [
+        Benchee.Formatters.Console,
+        {Benchee.Formatters.HTML, file: "samples_output/flat_map.html"}
+      ]
+    )
   """
 
-  @doc """
-  Transforms the statistical results from benchmarking to html to be written
-  somewhere, such as a file through `IO.write/2`.
+  @behaviour Benchee.Formatter
 
-  Returns a map from file name/path to file content along with formatter options.
-  """
-  @spec format(Suite.t()) :: {%{Suite.key() => String.t()}, map}
-  def format(suite) do
-    suite
-    |> default_configuration
-    |> do_format
-  end
+  require EEx
 
-  @default_filename "benchmarks/output/results.html"
-  @default_auto_open true
-  @default_inline_assets false
-  defp default_configuration(suite) do
-    opts =
-      suite.configuration.formatter_options
-      |> Map.get(:html, %{})
-      |> Map.put_new(:file, @default_filename)
-      |> Map.put_new(:auto_open, @default_auto_open)
-      |> Map.put_new(:inline_assets, @default_inline_assets)
-
-    updated_configuration = %Configuration{suite.configuration | formatter_options: %{html: opts}}
-    load_specs_for_versions()
-    %Suite{suite | configuration: updated_configuration}
-  end
+  alias Benchee.{
+    Configuration,
+    Conversion,
+    Formatters.JSON,
+    Suite,
+    Statistics,
+    Utility.FileCreation
+  }
 
-  defp load_specs_for_versions do
+  alias Conversion.{Duration, Count, DeviationPercent}
+
+  @doc """
+  Transforms the statistical results from benchmarking to html reports.
+
+  Returns a map from file name/path to file content.
+  """
+  @spec format(Suite.t(), map) :: %{Suite.key() => String.t()}
+  def format(
+        %Suite{
+          scenarios: scenarios,
+          system: system,
+          configuration: %Configuration{unit_scaling: unit_scaling}
+        },
+        opts
+      ) do
     _ = Application.load(:benchee)
     _ = Application.load(:benchee_html)
```

Comment:
by inlining these we los the speaking method name as to why we do this. I'd put them back in a function, alternatively add a coment (but as they are no major functionality I like hiding it in a function)